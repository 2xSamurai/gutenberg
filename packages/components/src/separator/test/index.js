/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { Separator } from '../index';

describe( 'props', () => {
	let base;

	beforeEach( () => {
		base = render( <Separator /> ).container;
	} );

	test( 'should render correctly', () => {
		expect( base.firstChild ).toMatchSnapshot();
	} );

	test( 'should render marginTop', () => {
		const { container } = render( <Separator marginTop={ 5 } /> );
		expect( container.firstChild ).toMatchStyleDiffSnapshot(
			base.firstChild
		);
	} );

	test( 'should render marginBottom', () => {
		const { container } = render( <Separator marginBottom={ 5 } /> );
		expect( container.firstChild ).toMatchStyleDiffSnapshot(
			base.firstChild
		);
	} );

	test( 'should render margin', () => {
		const { container } = render( <Separator margin={ 7 } /> );
		expect( container.firstChild ).toMatchStyleDiffSnapshot(
			base.firstChild
		);
	} );
} );
