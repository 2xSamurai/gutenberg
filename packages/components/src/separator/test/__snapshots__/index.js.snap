// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render correctly 1`] = `
.emotion-0 {
  border-color: rgba(0,0,0,0.1);
  border-width: 0 0 1px 0;
  height: 0;
  margin: 0;
  width: auto;
  margin-bottom: calc(4px * 3);
  margin-top: calc(4px * 3);
}

<hr
  aria-orientation="horizontal"
  class="components-divider components-separator emotion-0"
  data-wp-c16t="true"
  data-wp-component="Separator"
  role="separator"
/>
`;

exports[`props should render margin 1`] = `
Snapshot Diff:
- Received styles
+ Base styles

@@ -2,10 +2,10 @@
    Object {
      "border-color": "rgba(0,0,0,0.1)",
      "border-width": "0 0 1px 0",
      "height": "0",
      "margin": "0",
-     "margin-bottom": "calc(4px * 7)",
-     "margin-top": "calc(4px * 7)",
+     "margin-bottom": "calc(4px * 3)",
+     "margin-top": "calc(4px * 3)",
      "width": "auto",
    },
  ]
`;

exports[`props should render marginBottom 1`] = `
Snapshot Diff:
- Received styles
+ Base styles

@@ -2,10 +2,10 @@
    Object {
      "border-color": "rgba(0,0,0,0.1)",
      "border-width": "0 0 1px 0",
      "height": "0",
      "margin": "0",
-     "margin-bottom": "calc(4px * 5)",
+     "margin-bottom": "calc(4px * 3)",
      "margin-top": "calc(4px * 3)",
      "width": "auto",
    },
  ]
`;

exports[`props should render marginTop 1`] = `
Snapshot Diff:
- Received styles
+ Base styles

@@ -3,9 +3,9 @@
      "border-color": "rgba(0,0,0,0.1)",
      "border-width": "0 0 1px 0",
      "height": "0",
      "margin": "0",
      "margin-bottom": "calc(4px * 3)",
-     "margin-top": "calc(4px * 5)",
+     "margin-top": "calc(4px * 3)",
      "width": "auto",
    },
  ]
`;
